app/lib/account/backend.dart

```diff
@@ -179,16 +179,20 @@ class AccountBackend {
     return await _db.lookupValue<Like>(key, orElse: () => null);
   }
 
-  /// Return an iterable with the names of all the packages that the given [user] likes.
-  Future<List<String>> listPackageLikes(User user) async {
-    final likes = await _db.query<Like>(ancestorKey: user.key).run().toList();
-    return likes.map((Like l) => l.package).toList();
+  /// Returns a list with [LikeData] of all the packages that the given
+  ///  [user] likes.
+  Future<List<LikeData>> listPackageLikes(User user) async {
+    return await cache.userPackageLikes(user.userId).get(() async {
+      final query = _db.query<Like>(ancestorKey: user.key)..limit(1000);
```

Comment:
Please leave a TODO to introduce pagination and/or migrate this to regular search.

Reply:
A TODO about pagination would be fine.

Reply:
Done